griptape/processors/prompt_driver_processor.py

```diff
@@ -0,0 +1,24 @@
+from .base_processors import BasePromptStackProcessor
+
+
+class PromptDriverPiiProcessor(BasePromptStackProcessor):
```

Comment:
I think the idea behind `PromptDriverPiiProcessor` is to pass a prompt to a Prompt Driver that instructs the LLM to mask the text.

Reply:
@vasinov we would need to do a Prompt Driver `run` call on each Input in the Prompt Stack, right? This could get slow/expensive..

Reply:
One other approach could be to assemble the Prompt Stack into a prompt string, with clear separators between the Prompt Stack Inputs. Then we could make a single call to the LLM, and convert the response back to Prompt Stack Inputs.

Though this feels kind of hacky and might overwhelm LLMs that would otherwise be great at masking.

Reply:
another way could be to process the inputs in batches, rather than individually. This would reduce the number of function calls, which could potentially improve performance. Here's a rough idea of how this could be implemented.
```
def before_run(self, prompt_stack):
    batched_inputs = self.batch_inputs(prompt_stack["inputs"])
    for batch in batched_inputs:
        batch = self.mask_pii_func(batch)
    return prompt_stack

def after_run(self, prompt_stack):
    batched_inputs = self.batch_inputs(prompt_stack["inputs"])
    for batch in batched_inputs:
        batch = self.unmask_pii_func(batch)
    return prompt_stack 
```
Let me know what do you guys think

Reply:
Let's just get it working with the "slow" method for now and then we can optimize from there. Parallelization might be another approach if it comes down to it.

Reply:
> One other approach could be to assemble the Prompt Stack into a prompt string, with clear separators between the Prompt Stack Inputs.

We could convert it into a JSON object to make separation explicit. But parallelization is probably the better option.